java/util/src/main/java/com/google/protobuf/util/JsonFormat.java

```diff
@@ -1746,6 +1746,10 @@ private EnumValueDescriptor parseEnum(EnumDescriptor enumDescriptor, JsonElement
           // Fall through. This exception is about invalid int32 value we get from parseInt32() but
           // that's not the exception we want the user to see. Since result == null, we will throw
           // an exception later.
+            
+          if (ignoringUnknownFields) {
+            result = enumDescriptor.getValues().get(0);
```

Comment:
This will cause the field to have the default value, i.e., instead of ignoring unknown enum values, they are converted to the default value. It doesn't matter to proto3 because having the default value == not set, but for proto2 it means data corruption (users will get an incorrect enum value).

Could you find a way to ignore the field entirely instead?

Reply:
I actually based this on your comment dated Jun 30 2017 for the discussion on https://github.com/google/protobuf/pull/3289/commits/70049452c2ece7918b3bf8cb992b8178e6e5e1cf
for veerenmandalia's implementation

Having to "treat it as if the field is not present" would mean assigning it a default value.

Reply:
"treat it as if the field is not present" does not mean assigning it a default value. In proto3, assigning default value == not present, but that's not the case in proto2. Here we need to find a way to skip the field entirely and not set any value into the field.

Reply:
Ok. So instead, if i return null from this method, the protobuf field will be not be set. In my test, that is also returning the default value as the builder enum int is preset to 0 and the proto generated class returns default if the enum int value is 0. Should returning null be the fix here? How will it behave with proto2 compiler?

Reply:
I think the code needs some refactoring to be able to ignore unknown enum values. Returning null is one option and we need to update all callsites of the method to make sure they are handling the null return value properly.

Reply:
I have made some refactoring change. I also saw the call sites where the changed method is called. In map, it is adding the null as value. As a result, it should return a default 0th enum. Accordingly I have added 2 more tests, one for repeated enum and other for enum in map value.